docs/robots/jackal/tutorials_jackal.mdx

```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
+You can use the `apt` package management system to receive new versions all software running on the platform.
+
+Each Jackal leaves the factory already configured to pull packages from http://packages.ros.org as well as http://packages.clearpathrobotics.com.
+To update your package and download new package versions, simply run:
+
+```
+sudo apt-get update
+sudo apt-get dist-upgrade
+```
+
+#### MCU Firmware Update
+
+:::note
+
+Jackal's MCU is mounted to the rear inside wall of the robot. To access it, open the lid, keeping the computer
+tray secured to the underside of the lid.
+
+<center>
+  <figure>
+    <img
+      src="/img/robot_images/jackal_images/mcu-annotated.jpg"
+      width="800"
+    />
+    <figcaption>Jackal's MCU</figcaption>
+  </figure>
+</center>
+
+1. The `M_RST` button, used to reset the MCU.
+2. The MCU mode switch. Left is `NORM` or normal mode, right is `ALT` used for flashing the firmware.
+3. The mini-USB port used for transmitting data. This should be connected to Jackal's PC for both normal operation and firmware flashing.
+
+:::
+
+When you update packages, there is periodically a new version of Jackal's firmware available. You will know this
+is the case in one of two ways:
+
+1. The firmware and PC are unable to connect to each other, which will be apparent if the two-arrow comms indicator
+   fails to come on after system bootup.
+2. The firmware version number in the `/status` message does not match the package version output by
+   `dpkg -s ros-noetic-jackal-firmware`. In the future there will be an automated check for this which outputs
+   a diagnostics warning when a firmware update is available to be applied.
+
+If new firmware is available, follow the procedure below to flash it to Jackal's MCU:
+
+1. Place Jackal up on blocks. Firmware loading does not usually result in unintended motion, but it's safest when
+   off the ground.
+2. Ensure that Jackal is powered-on. Open the lid of the robot to expose the MCU.
+3. Make sure the mini-USB cable (`3` in the image above) is connected to Jackal's PC.
+4. Move the mode switch (`2` in the image above) to `ALT` (right).
+5. Press `M_RST` (`1` in the image above) to enter boot loader mode.
+6. The MCU is now in bootloader mode, ready to receive new firmware.
+
+Now, from Jackal's PC (connected via SSH or screen/keyboard), execute:
+
+```
+rosrun jackal_firmware upload
+```
+
+You should see about 20 seconds worth of lines output beginning with `Download from image ...`. When this is
+complete perform the following steps to restore normal operation:
+
+1. Move the mode switch back to `NORM` (left).
+2. Press `M_RST`. You must perform this step within a few seconds of changing the mode switch. If you take too long Jackal may power down. This is fine and won't harm the robot, but can be annoying.
+
+If the robot powers-down, simply ensure the mode switch is in the `NORM` position and power the robot back on.
+
+Jackal will now be using the new firmware you installed.
```

Comment:
Should we mention moving MCU switches back to their original positions, or rebooting the robot?

Reply:
I didn't quite follow what the recommended change was here. Are there switches other than the "NORM" position that need to be set?

Reply:
My mistake. You have told the User to move the switches back to the original positions:
`1. Move the mode switch back to `NORM` (left).`